SSNFSd/ssnfsworker.cpp

```diff
@@ -395,13 +413,29 @@ void SSNFSWorker::ReadyToRead()
             qint32 uid = Common::getInt32FromBytes(Common::readExactBytes(socket, 4));
 
             quint32 pathLength = Common::getUInt32FromBytes(Common::readExactBytes(socket, 4));
-
-            QByteArray targetPath = Common::readExactBytes(socket, pathLength);
+            // Returns QString::null if the path is not valid
+            QByteArray targetPath = isValidPath(Common::readExactBytes(socket, pathLength));
 
             int res;
 
             struct stat stbuf;
 
+            if (targetPath.isNull()){
```

Comment:
If the function is changed to return a boolean, it can just be called from the `if`.